Sources/NIOSSL/SecurityFrameworkCertificateVerification.swift

```diff
@@ -47,11 +47,40 @@ extension SSLConnection {
             var trust: SecTrust? = nil
             var result: OSStatus
             let policy = SecPolicyCreateSSL(self.role! == .client, self.expectedHostname as CFString?)
-            result = SecTrustCreateWithCertificates(certificates as CFArray, policy, &trust)
+            result = SecTrustCreateWithCertificates(peerCertificates as CFArray, policy, &trust)
             guard result == errSecSuccess, let actualTrust = trust else {
                 throw NIOSSLError.unableToValidateCertificate
             }
 
+            // If there are additional trust roots then we need to add them to the SecTrust as anchors.
+            let additionalAnchorCertificates: [SecCertificate] = try self.parentContext.configuration.additionalTrustRoots.flatMap { trustRoots -> [NIOSSLCertificate] in
+                switch trustRoots {
+                case .file(let path):
+                    return try NIOSSLCertificate.fromPEMFile(path)
```

Comment:
This branch is untakeable, as we did the conversion earlier. We can `preconditionFailure` here.

Reply:
OK, I designed it to not be coupled with the other code that did the conversion and saw the conversion in `SSLContext` to be an optimisation.

But I can fail hard here if we want to make it a requirement for the user of this callback to have done any conversion?

Reply:
I think that would be good, it clarifies intended flow.

Reply:
Addressed in https://github.com/apple/swift-nio-ssl/pull/284/commits/9d613d303a6623c384855ea67b52af5a318aeea6.